File: backend/schemas/draft.py
from pydantic import BaseModel
from datetime import date, datetime
from schemas.position import PositionResponse


class DraftBase(BaseModel):
    salary_code: int
    preset: int
    start_period: date
    end_period: date


class DraftCreate(DraftBase):
    position_id: str


class DraftUpdate(BaseModel):
    pass


class Draft(DraftBase):
    id: str
    team_id: str
    create_date: datetime
    position: PositionResponse


class DraftForContract(DraftBase):
    id: str
    team_id: str
    team_name: str
    position_id: str
    unit_pay: int

File: frontend/src/types/draft.ts
import { SalaryType } from './employee';
import { PositionData } from './position';

/** 그룹 데이터 인터페이스 */
export interface DraftData {
  id: string;
  salaryCode: SalaryType;
  preset: number;
  startPeriod: string;
  endPeriod: string;
  createDate: string;
  position: PositionData;
}

/** 그룹 생성 API 바디 */
export type DraftCreateBody = {
  positionId: string;
} & Omit<DraftData, 'id' | 'createDate' | 'position'>;
